libsodium decryption panics (index out of range [0] with length 0)

### Database name
PostgreSQL

# Issue description
libsodium decryption panics.

### Describe your problem
Backup restoration fails when using libsodium encryption. 

### Steps to reproduce
0) Create a backup
1) Restore a backup (i.e. wal-g backup-restore /tmp/foo LATEST)

### Output
```bash
# wal-g backup-restore ... ...
panic: runtime error: index out of range [0] with length 0

goroutine 84 [running]:
github.com/wal-g/wal-g/internal/crypto/libsodium.(*Reader).readNextChunk(0xc001393680)
        /build/internal/crypto/libsodium/reader.go:108 +0x40f
github.com/wal-g/wal-g/internal/crypto/libsodium.(*Reader).Read(0xc001393680, {0xc000550a20, 0x4, 0xc001393680})
        /build/internal/crypto/libsodium/reader.go:78 +0xb7
io.ReadAtLeast({0x1ae30c0, 0xc001393680}, {0xc000550a20, 0x4, 0xf}, 0x4)
        /usr/local/go/src/io/io.go:328 +0x9a
io.ReadFull(...)
        /usr/local/go/src/io/io.go:347
github.com/pierrec/lz4/v4/internal/lz4stream.(*Frame).readUint32(0xc000550a20, {0x1ae30c0, 0xc001393680})
        /go/pkg/mod/github.com/pierrec/lz4/v4@v4.1.11/internal/lz4stream/block.go:343 +0x45
github.com/pierrec/lz4/v4/internal/lz4stream.(*Frame).CloseR(0xc000550a20, {0x1ae30c0, 0xc001393680})
        /go/pkg/mod/github.com/pierrec/lz4/v4@v4.1.11/internal/lz4stream/frame.go:124 +0x55
github.com/pierrec/lz4/v4.(*Reader).Read(0xc00144c140, {0xc0013d0000, 0x14d29c0, 0x11a6201})
        /go/pkg/mod/github.com/pierrec/lz4/v4@v4.1.11/reader.go:133 +0x54e
github.com/wal-g/wal-g/internal.readTrailingZeros({0x7f8d3c14d968, 0xc00144a240})
        /build/internal/extract.go:253 +0x62
github.com/wal-g/wal-g/internal.extractFile({0x1ae3120, 0xc000bba210}, {0x7f8d3c14d968, 0xc00144a240}, {0x1b138b8, 0xc000bc0fc0})
        /build/internal/extract.go:181 +0xcf
github.com/wal-g/wal-g/internal.tryExtractFiles.func1()
        /build/internal/extract.go:222 +0x247
created by github.com/wal-g/wal-g/internal.tryExtractFiles
        /build/internal/extract.go:210 +0x30f
```
### Config
```
WALG_GS_PREFIX=gs://my-bucket/my-folder
GOOGLE_APPLICATION_CREDENTIALS=/run/secrets/gcs-credentials
WALG_LIBSODIUM_KEY=0000000000000000000000000000000000000000000000000000000000000000
WALG_LIBSODIUM_KEY_TRANSFORM=hex
PGHOST=127.0.0.1
PGPORT=5432
PGUSER=postgres
PGPASS=mysecretpassword
```

--- a/internal/crypto/libsodium/crypter_test.go
+++ b/internal/crypto/libsodium/crypter_test.go
@@ -71,6 +71,12 @@ func EncryptionCycle(t *testing.T, crypter crypto.Crypter) {
 	assert.NoErrorf(t, err, "decryption read error: %v", err)
 
 	assert.Equal(t, secret, string(decrypted), "decrypted text not equals to open text")
+
+	// decrypter should keep returning EOF with no data
+	var buf [8]byte
+	n, err := decrypt.Read(buf[:])
+	assert.Equal(t, n, 0, "decryptor should not read any more data after ReadAll")
+	assert.Error(t, io.EOF, "decryptor should keep returning EOF error")
 }
 
 func TestEncryptionCycleFromKey(t *testing.T) {

--- a/internal/crypto/libsodium/reader.go
+++ b/internal/crypto/libsodium/reader.go
@@ -88,8 +88,6 @@ func (reader *Reader) Read(p []byte) (n int, err error) {
 func (reader *Reader) readNextChunk() (err error) {
 	n, err := io.ReadFull(reader.Reader, reader.in)
 
-	reader.in = reader.in[:n]
-
 	if err != nil && err != io.ErrUnexpectedEOF {
 		return
 	}

